Issues building library package for CentOS/RHEL 8

My first attempt at building a library package for ziti sdk shared library fails around libsodium.
Since the distribution has already libsodium, I have installed `libsodium-devel` and essentially running `cmake` with `-DHAVE_LIBSODIUM=ON` however that fails with:

> Unknown arguments specified

Whether I pass it as a cmd line option to make or change the option to default to `ON` in CMakeLists.txt, the same thing happens.

Here's entire RPM build log starting from `prep` section where I do the necessary things in order to built from a github archive..

```
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.UyXgJe
+ umask 022
+ cd /builddir/build/BUILD
+ cd /builddir/build/BUILD
+ rm -rf ziti-sdk-c-0.22.0
+ /usr/bin/gzip -dc /builddir/build/SOURCES/0.22.0.tar.gz
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd ziti-sdk-c-0.22.0
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ sed -i 's@DESTINATION lib@DESTINATION lib64@' CMakeLists.txt
+ sed -i /git.cmake/d CMakeLists.txt
+ sed -ri 's@.*HAVE_LIBSODIUM.*@option(HAVE_LIBSODIUM "use and link installed shared libsodium library" ON)@' CMakeLists.txt
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.LmHtwl
+ umask 022
+ cd /builddir/build/BUILD
+ cd ziti-sdk-c-0.22.0
+ export ver=0.22.0
+ ver=0.22.0
+ mkdir ./build
~/build/BUILD/ziti-sdk-c-0.22.0/build ~/build/BUILD/ziti-sdk-c-0.22.0
+ pushd ./build
+ cmake3 .. -DCMAKE_BUILD_TYPE=Release '-DCMAKE_CXX_FLAGS:STRING=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' '-DCMAKE_C_FLAGS:STRING=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wno-error=unused-result' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_LIBDIR:PATH=/usr/lib64
CMake Warning at CMakeLists.txt:4 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


-- The C compiler identification is GNU 8.3.1
-- The CXX compiler identification is GNU 8.3.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
project version: 
git info:
   branch : 
     hash : 
          date : 

using Unix Makefiles
cross-compiling FALSE
Project: ziti-sdk@v
using mbedTLS libraries
CMake Warning (dev) at build/_deps/mbedtls-src/CMakeLists.txt:5 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Python3: /usr/bin/python3.6 (found version "3.6.8") found components: Interpreter 
uvlink src = /builddir/build/BUILD/ziti-sdk-c-0.22.0/build/_deps/uv_link-src
CMake Error at CMakeLists.txt:73 (if):
  if given arguments:

    "STREQUAL" "main"

  Unknown arguments specified


-- Configuring incomplete, errors occurred!
See also "/builddir/build/BUILD/ziti-sdk-c-0.22.0/build/CMakeFiles/CMakeOutput.log".


RPM build errors:
error: Bad exit status from /var/tmp/rpm-tmp.LmHtwl (%build)
    Bad exit status from /var/tmp/rpm-tmp.LmHtwl (%build)
Finish: rpmbuild ziti-0.22.0-1.el8.src.rpm
Finish: build phase for ziti-0.22.0-1.el8.src.rpm
ERROR: Exception(ziti-0.22.0-1.el8.src.rpm) Config(gps-8-x86_64) 0 minutes 32 seconds
INFO: Results and/or logs in: .
ERROR: Command failed: 
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/ziti.spec
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,8 +48,6 @@ if (DEFINED ENV{BUILD_NUMBER})
     set(ZITI_BUILDNUM $ENV{BUILD_NUMBER})
 endif ()
 
-set(ZITI_VERSION ${PROJECT_VERSION}-${ZITI_BUILDNUM})
-
 link_directories(${CMAKE_BINARY_DIR}/lib)
 
 add_subdirectory(deps)
@@ -67,41 +65,6 @@ elseif (WIN32)
 endif ()
 
 include(CPack)
-add_custom_target(publish
-        COMMENT "generate publish specs"
-        )
-
-if(${GIT_BRANCH} STREQUAL main)
-    set(PUB_PFX ziti-staging)
-else()
-    set(PUB_PFX ziti-snapshot/${GIT_BRANCH})
-endif()
-
-set(PUB_OPTS --apikey=$ENV{JFROG_API_KEY} --url https://netfoundry.jfrog.io/netfoundry --build-name=ziti --build-number=${ZITI_VERSION})
-string(TOLOWER ${CMAKE_SYSTEM_NAME} PUB_OS)
-string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} PUB_ARCH)
-
-if(${PUB_ARCH} STREQUAL "x86_64" )
-    set(PUB_ARCH "amd64")
-endif()
-
-macro(PUBCOMP comp)
-    set(${comp}_PUB_PATH ${PUB_PFX}/${comp}/${PUB_ARCH}/${PUB_OS}/${ZITI_VERSION}/${comp}.${archive_sfx})
-    set(${comp}_PUB_PROPS "version=${ZITI_VERSION}\;name=${comp}\;arch=${PUB_ARCH}\;os=${PUB_OS}\;branch=${GIT_BRANCH}")
-
-    add_custom_target(${comp}-pub
-            BYPRODUCTS ${CMAKE_BINARY_DIR}/${comp}-pub-spec.json
-            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${comp}-${ZITI_VERSION}-${CPACK_SYSTEM_NAME}.${archive_sfx}
-            COMMAND echo "{"
-            \"target\": \"${${comp}_PUB_PATH}\",
-            \"pattern\": \"${CMAKE_CURRENT_BINARY_DIR}/${comp}-${ZITI_VERSION}-${CPACK_SYSTEM_NAME}.${archive_sfx}\",
-            \"props\": \"${${comp}_PUB_PROPS}\"
-            "}" > ${CMAKE_BINARY_DIR}/${comp}-pub-spec.json
-            VERBATIM
-            )
-
-    add_dependencies(publish ${comp}-pub)
-endmacro(PUBCOMP)
 
 # use prefix length to trim path for logging, see utils.c
 if (CMAKE_GENERATOR STREQUAL "Ninja" AND CMAKE_BINARY_DIR MATCHES "${CMAKE_SOURCE_DIR}")
@@ -115,9 +78,7 @@ endif()
 add_subdirectory(library)
 
 if (ZITI_BUILD_PROGRAMS)
-    if (NOT ${PUB_OS} STREQUAL "ios")
-        add_subdirectory(programs)
-    endif ()
+    add_subdirectory(programs)
 endif ()
 
 if (ZITI_BUILD_TESTS)

--- a/git.cmake
+++ b/git.cmake
@@ -1,18 +1,51 @@
 # Get the current working branch
 execute_process(
-        COMMAND git rev-parse --abbrev-ref HEAD
+        COMMAND git status
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        OUTPUT_VARIABLE GIT_BRANCH
+        RESULT_VARIABLE GIT_STATUS
+        ERROR_QUIET
+        OUTPUT_QUIET
         OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 
-# Get the latest abbreviated commit hash of the working branch
-execute_process(
-        COMMAND git log -1 --format=%h
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        OUTPUT_VARIABLE GIT_COMMIT_HASH
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-)
+if (GIT_STATUS EQUAL 0)
+    message("getting version from git")
+
+    execute_process(
+            COMMAND git rev-parse --abbrev-ref HEAD
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+            OUTPUT_VARIABLE GIT_BRANCH
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # Get the latest abbreviated commit hash of the working branch
+    execute_process(
+            COMMAND git log -1 --format=%h
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+            OUTPUT_VARIABLE GIT_COMMIT_HASH
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # lookup most recent tag to derive version
+    execute_process(
+            COMMAND git describe --tags HEAD
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+            OUTPUT_VARIABLE GIT_INFO
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    if(${GIT_INFO} MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)$")
+        set(ver ${GIT_INFO})
+    elseif(${GIT_INFO} MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)-([0-9]+)-[^-]*")
+        string(JOIN "." ver ${CMAKE_MATCH_1} ${CMAKE_MATCH_2})
+    endif()
+    message("project version: ${ver} (derived from git)")
+else()
+    message("no git information available\ntrying to set version from ZITI_SDK_VERSION environment var")
+    set(ver $ENV{ZITI_SDK_VERSION})
+    set(GIT_COMMIT_HASH "")
+    set(GIT_BRANCH "unknown")
+endif()
 
 # capture build date
 if (WIN32)
@@ -30,20 +63,6 @@ execute_process(
 )
 endif()
 
-# lookup most recent tag to derive version
-execute_process(
-        COMMAND git describe --tags HEAD
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        OUTPUT_VARIABLE GIT_INFO
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-if(${GIT_INFO} MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)$")
-    set(ver ${GIT_INFO})
-elseif(${GIT_INFO} MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)-([0-9]+)-[^-]*")
-    string(JOIN "." ver ${CMAKE_MATCH_1} ${CMAKE_MATCH_2})
-endif()
-message("project version: ${ver} (derived from git)")
 
 
 

--- a/programs/ziti-prox-c/CMakeLists.txt
+++ b/programs/ziti-prox-c/CMakeLists.txt
@@ -1,4 +1,3 @@
-project(ziti-prox-c)
 
 add_executable(ziti-prox-c proxy.c)
 
@@ -10,7 +9,8 @@ endif()
 target_link_libraries(ziti-prox-c PUBLIC ziti subcommand)
 target_include_directories(ziti-prox-c PRIVATE ${ziti-sdk_SOURCE_DIR}/inc_internal)
 
-set(ZITI_ARCHIVE_NAME "ziti-prox-c-${ZITI_VERSION}-${CPACK_SYSTEM_NAME}.${archive_sfx}")
+set(ZITI_ARCHIVE_NAME "ziti-prox-c-${PROJECT_VERSION}-${CPACK_SYSTEM_NAME}.${archive_sfx}")
+message(${ZITI_ARCHIVE_NAME})
 set(ZITI_EXECUTABLE_NAME "ziti-prox-c${CMAKE_EXECUTABLE_SUFFIX}")
 if (UNIX)
     add_custom_target(ziti-prox-c-pack
@@ -25,6 +25,3 @@ else()
         COMMAND zip "${ZITI_ARCHIVE_NAME}" "${PROJECT_BINARY_DIR}/${CMAKE_BUILD_TYPE_INIT}/${ZITI_EXECUTABLE_NAME}"
     )
 endif()
-
-#add_dependencies(publish ${ZITI_ARCHIVE_NAME})
-PUBCOMP(ziti-prox-c)

--- a/programs/ziti-prox-c/CMakeLists.txt
+++ b/programs/ziti-prox-c/CMakeLists.txt
@@ -10,7 +10,6 @@ target_link_libraries(ziti-prox-c PUBLIC ziti subcommand)
 target_include_directories(ziti-prox-c PRIVATE ${ziti-sdk_SOURCE_DIR}/inc_internal)
 
 set(ZITI_ARCHIVE_NAME "ziti-prox-c-${PROJECT_VERSION}-${CPACK_SYSTEM_NAME}.${archive_sfx}")
-message(${ZITI_ARCHIVE_NAME})
 set(ZITI_EXECUTABLE_NAME "ziti-prox-c${CMAKE_EXECUTABLE_SUFFIX}")
 if (UNIX)
     add_custom_target(ziti-prox-c-pack

